core/server/api_container/server/startosis_engine/kurtosis_types/service_config/service_config.go

```diff
@@ -678,32 +679,30 @@ func convertImage(
 	packageId string,
 	packageContentProvider startosis_packages.PackageContentProvider,
 	packageReplaceOptions map[string]string) (string, *image_build_spec.ImageBuildSpec, *image_registry_spec.ImageRegistrySpec, *nix_build_spec.NixBuildSpec, *startosis_errors.InterpretationError) {
-	imageBuildSpecStarlarkType, isImageBuildSpecStarlarkType := image.(*ImageBuildSpec)
-	imageSpecStarlarkType, isImageRegistrySpecStarlarkType := image.(*ImageSpec)
-	nixBuildSpecStarlarkType, isNixBuildSpecStarlarkType := image.(*NixBuildSpec)
-	if isImageBuildSpecStarlarkType {
-		imageBuildSpec, interpretationErr := imageBuildSpecStarlarkType.ToKurtosisType(locatorOfModuleInWhichThisBuiltInIsBeingCalled, packageId, packageContentProvider, packageReplaceOptions)
+	switch image := image.(type) {
```

Comment:
turned this into a switch statement as opposed to the `if elses` that wre getting really messy, but image type should be refactored to a common type somehow